app/uk/gov/hmrc/tradergoodsprofilesrouter/controllers/action/ValidationRules.scala

```diff
@@ -114,6 +114,7 @@ object ValidationRules {
 
   private val actorIdPattern: Regex = raw"[A-Z]{2}\d{12,15}".r
   private val comcodePattern: Regex = raw".{6}(.{2}(.{2})?)?".r
+  private val niphlsPattern: Regex  = """^\d{4}$|^\d{5}$|^\d{6}$|^[A-Za-z]\d{5}$|^[A-Za-z]\d{6}$""".r
```

Comment:
You need to include 7 characters in this check as well

Reply:
I also think you could simplify the regex with this: 
^([a-zA-Z0-9]){4,7}$

Reply:
The current regex accepts 7 characters like this S123456. The suggested regex accepts S1234 - Alpha-numeric-5, and S123 - Alpha-numeric-4, which are not part of what the ticket specified, if those are acceptable, we need to update the ticket as well to describe all possible acceptable patterns